base/abstractarray.jl

```diff
@@ -888,10 +888,11 @@ julia> empty([1.0, 2.0, 3.0], String)
 String[]
 ```
 """
-empty(a::AbstractVector{T}, ::Type{U}=T) where {T,U} = Vector{U}()
+empty(a::AbstractVector{T}, ::Type{U}=T) where {T,U} = similar(a, U, 0)
 
-# like empty, but should return a mutable collection, a Vector by default
-emptymutable(a::AbstractVector{T}, ::Type{U}=T) where {T,U} = Vector{U}()
+# like empty, but should return a mutable collection
+emptymutable(a::AbstractVector{T}, ::Type{U}=T) where {T,U} = similar(a, U, 0)
```

Comment:
L891 looks OK to me.
But I'm not sure if L894 is breaking. (`emptymutable` is used in `union` and the general fallback is `push!` based.)

Reply:
I've reverted the `emptymutable` change, so only L891 remains now